src/pixelator/pna/config/panel.py

```diff
@@ -97,6 +101,82 @@ def from_csv(cls, filename: PathType) -> Self:
 
         return cls(df, metadata, file_name=panel_file.name)
 
+    @classmethod
+    def from_pxl(cls, filename: PathType) -> Self:
+        """Create an AntibodyPanel from a pxl file.
+
+        :param filename: The path to the panel file.
+        :returns: The AntibodyPanel object.
+        :raises AssertionError: exception if panel file is missing,
+        :rtype: AntibodyPanel
+        """
+        pxl_file = Path(filename)
+
+        if not pxl_file.is_file() or pxl_file.suffix != ".pxl":
+            raise AssertionError(
+                f"Panel file {filename} not found or has an incorrect format"
+            )
+
+        logger.debug("Creating Antibody panel from file %s", filename)
+
+        pxl_data = read_pna(pxl_file)
+        adata = pxl_data.adata()
+        df = adata.var.reset_index(names="marker_id")[cls._REQUIRED_COLUMNS]
+        if "uniprot_id" in adata.var.columns:
+            df.insert(1, "uniprot_id", adata.var.reset_index()["uniprot_id"])
```

Comment:
I'm wondering if perhaps we need to think a little bit about how we deal with this - i.e. when we add more types of metadata to the panel file, how do we deal with that. Ideally we want everything that was in the original panel file to be added to `var`, right? One idea I had now is that we could add a column to `var` that says something like `info_from_panel_file` and then use that to determine what to feed forward here. Does that make sense?

Reply:
You mean we want the code to be more agnostic to the format of the panel data, right?

I'm not sure I see how such a column would look like, but alternatively we could have an extra field in the panel_metadata that would contain the csv header of the panel?